### PR TITLE
OCPBUGS-44961: Fix "Enforce the required-scc monitor test and validate usage of non-standard OCP SCCs"

### PR DESCRIPTION
**Pull request by kramaranya**
Enforced required-scc monitor test for fixed namespaces. Extended the required-scc monitor test to validate non-standard SCCs and ensure they are only used in allowed namespaces.

Fixed the issue from revert #29321 

Added system namespaces such as `default`, `kube-system`, `kube-public`, `openshift-node`, `openshift-infra`, and `openshift` to ensure they are treated as flaking namespaces.

### CODE REVIEW

File: pkg/monitortests/authentication/requiredsccmonitortests/monitortest.go [hunk 1]
```diff
@@ -31,6 +31,35 @@ var defaultSCCs = sets.NewString(
...
+	"openshift-cluster-version",
+	"openshift-image-registry",
+	"openshift-ingress",
+	"openshift-ingress-canary",
+	"openshift-ingress-operator",
+	"openshift-insights",
+	"openshift-machine-api",
+	"openshift-marketplace",
+	"openshift-monitoring",
+)
```
Comment (stbenjam) [inline pkg/monitortests/authentication/requiredsccmonitortests/monitortest.go:50]: Can you include https://github.com/openshift/origin/blob/master/test/extended/util/managed_services.go#L8-L28 in this set?

`.Union(ManagedServiceNamespaces)`

File: pkg/monitortests/authentication/requiredsccmonitortests/monitortest.go [hunk 2]
```diff
@@ -61,27 +112,70 @@ func (w *requiredSCCAnnotationChecker) CollectData(ctx context.Context, storageD
 
 	junits := []*junitapi.JUnitTestCase{}
 	for _, ns := range namespaces.Items {
-		// require that all workloads in openshift, kube-* or default namespaces must have the required-scc annotation
+		// skip managed service namespaces
+		if exutil.ManagedServiceNamespaces.Has(ns.Name) {
+			continue
+		}
```
Comment (kramaranya) [inline pkg/monitortests/authentication/requiredsccmonitortests/monitortest.go:118]: @stbenjam we have decided to skip scc pinning for the namespaces in `ManagedServiceNamespaces`. If you believe these namespaces should instead be treated as flaking, please let us know.

File: pkg/monitortests/authentication/requiredsccmonitortests/monitortest.go [hunk 3]
```diff
@@ -61,27 +112,70 @@ func (w *requiredSCCAnnotationChecker) CollectData(ctx context.Context, storageD
...
+
+		// require that all workloads in openshift, kube-*, or default namespaces must have the required-scc annotation
 		// ignore openshift-must-gather-* namespaces which are generated dynamically
 		isPermanentOpenShiftNamespace := (ns.Name == "openshift" || strings.HasPrefix(ns.Name, "openshift-")) && !strings.HasPrefix(ns.Name, "openshift-must-gather-")
 		if !strings.HasPrefix(ns.Name, "kube-") && ns.Name != "default" && !isPermanentOpenShiftNamespace {
 			continue
 		}
 
+		// check if the namespace should be treated as flaking when failed
+		flakeWhenFailed := namespacesWithPendingSCCPinning.Has(ns.Name) ||
```
Comment (liouk) [inline pkg/monitortests/authentication/requiredsccmonitortests/monitortest.go:128]: The condition is quite simple, no need for a variable.

**Review by liouk — CHANGES_REQUESTED** [review]

File: pkg/monitortests/authentication/requiredsccmonitortests/monitortest.go (hunk 2 above)
Comment (kramaranya) [reply pkg/monitortests/authentication/requiredsccmonitortests/monitortest.go:118]: FYI ^^^ Flaking means that the test might sometimes pass and sometimes fail. By treating these namespaces as flaking, we allow our tests to continue without blocking progress. So if we decide to treat them as flaking, those namespaces should be fixed and eventually removed from the flaking list.

**Comment by liouk** [comment]
Holding for open question: https://github.com/openshift/origin/pull/29323#discussion_r1858367457

/hold

File: pkg/monitortests/authentication/requiredsccmonitortests/monitortest.go (hunk 2 above)
Comment (stbenjam) [reply pkg/monitortests/authentication/requiredsccmonitortests/monitortest.go:118]: Skipping them entirely sounds fine, thanks

**Comment by kramaranya** [comment]
/payload-job periodic-ci-openshift-hypershift-release-4.19-periodics-e2e-aws-ovn-conformance

**Pull request merged**